bin/testEnvLocalK8S.sh

```diff
@@ -128,49 +133,82 @@ function startLocalApiserver() {
     # make sure apiserver is actually alive
     kill -0 $(cat $LOG_DIR/apiserver.pid)
 
-    echo "Started local etcd and apiserver !"
+    # Really need to make sure that API Server is up before proceed further
+    waitForApiServer
+    printf "Started local etcd and apiserver!\n"
+}
+
+function ensureLocalApiServer() {
+    kubectl get nodes 2>/dev/null || startLocalApiserver
+}
+
+function createIstioConfigmap() {
+  helm template ${ISTIO_GO}/install/kubernetes/helm/istio --namespace=istio-system \
+     --execute=templates/configmap.yaml --values install/kubernetes/helm/istio/values.yaml  > /tmp/istio-configmap.yaml
+  kubectl create -f /tmp/istio-configmap.yaml 
+  helm template ${ISTIO_GO}/install/kubernetes/helm/istio --namespace=istio-system \
+     --execute=charts/ingress/templates/service.yaml --values install/kubernetes/helm/istio/values.yaml  > /tmp/istio-ingress.yaml
```

Comment:
this /tmp dir usage looks a little problematic from security POV.  I would take a look at man mktemp(1) page has example usage.  I don't know the context in which this is used (is it in the gate, on a local dev environment, on a shared machine?)

Depending on the answer it may not matter.  It should be relatively easy to make a secure file in tmp with mktemp

Reply:
I will use istio's LOG_DIR variable for the location of temp files.